File: 2544-alternating-digit-sum/2544-alternating-digit-sum.cpp
class Solution {
public:
    int alternateDigitSum(int n) {
        vector<int> digs;
        for(; n > 0; n /= 10) {
            digs.push_back(n % 10);
        }
        int ans = 0;
        int sgn = 1;
        for(auto it = digs.rbegin(); it != digs.rend(); ++it) {
            ans += sgn * (*it);
            sgn = -sgn;
        }
        
        return ans;
    }
};

File: 2544-alternating-digit-sum/2544-alternating-digit-sum.java
class Solution:
    def alternateDigitSum(self, n: int) -> int:
        digits = []

        while n:
            digits.append(n%10)
            n //= 10
        
        return sum(digits[-1-i] * (1 if i%2 == 0 else -1) for i in range(len(digits)))